CMakeLists.txt

```diff
@@ -44,16 +44,16 @@ if(USE_CUDA AND NOT USE_OLDCMAKECUDA)
         OR (${CMAKE_GENERATOR} STREQUAL "Unix Makefiles")
       ) AND (
         (${CMAKE_VERSION} VERSION_GREATER "3.9.0") OR (${CMAKE_VERSION} VERSION_EQUAL "3.9.0")
-      )
+      ) AND NOT USE_OLDCMAKECUDA
```

Comment:
I think it makes more sense up top in the old place because this else block just reset it again to false, right?

Reply:
else block set "FIRST_CUDA" to false,not set "USE_OLDCMAKECUDA".

Once the FIRST_CUDA is set up, it doesn't need to be checked USE_OLDCMAKECUDA

Reply:
It can also be in the old place..